src/devices/cpu/tlcs870/tlcs870.h

```diff
@@ -124,53 +108,234 @@ class tlcs870_device : public cpu_device
 	int     m_icount;
 
 	// Work registers
-	uint16_t m_op;
-	int m_param2_type;
-	uint16_t m_param2;
-
-	int m_param1_type;
-	uint16_t m_param1;
-	uint16_t m_temppc; // this is just PPC? use generic reg?
-
-	uint8_t m_bitpos;
-	uint8_t m_flagsaffected;
 	uint8_t m_cycles;
-
+	uint16_t m_tmppc;
 	uint32_t  m_addr;
 
 	uint16_t m_F;
 
 	/* CPU registers */
 	uint8_t m_RBS; // register base (4-bits)
 
-	inline uint8_t  RM8 (uint32_t a);
-	inline uint16_t RM16(uint32_t a);
-	inline void WM8 (uint32_t a, uint8_t  v);
-	inline void WM16(uint32_t a, uint16_t v);
-	inline uint8_t  RX8 (uint32_t a, uint32_t base);
-	inline uint16_t RX16(uint32_t a, uint32_t base);
-	inline void WX8 (uint32_t a, uint8_t  v, uint32_t base);
-	inline void WX16(uint32_t a, uint16_t v, uint32_t base);
-	inline uint8_t  READ8();
-	inline uint16_t READ16();
-	void decode();
-	void decode_register_prefix(uint8_t b0);
-	void decode_source(int type, uint16_t val);
-	void decode_dest(uint8_t b0);
+	inline uint8_t  RM8(uint32_t a) { return m_program->read_byte(a); }
+	inline uint16_t RM16(uint32_t a) { return RM8(a) | (RM8((a + 1) & 0xffff) << 8); }
+
+	inline void WM8(uint32_t a, uint8_t  v) { m_program->write_byte(a, v); }
+	inline void WM16(uint32_t a, uint16_t v) { WM8(a, v);    WM8((a + 1) & 0xffff, v >> 8); }
```

Comment:
It seems to be a little-known fact, but if you supply the body of the member function at declaration point like this, `inline` is implied - there's no need to explicitly mark them inline.  It makes the lines a bit shorter and less cluttered if you leave out the `inline`.